api/validate.py

```diff
@@ -63,3 +65,20 @@ def validate_string(req: Request, field: str, *, required: bool = False) -> str:
     if value == "" and required:
         raise ValueError(f"Required parameter '{field}' is missing")
     return value
+
+
+def validate_lang(
+    req: Request, field: str, *, default: str = "en-us", required: bool = False
+) -> str:
+    """Validate a string with a locale lang, returns the string if valid, otherwise the default.
+
+    Raises ValueError if the field is required and the valid is not valid.
+    """
+    value = req.args.get(field, "")
+    locales = ["en-us", "fr"]
+    if value == "" and required:
+        raise ValueError(f"Required parameter '{field}' is missing")
+    elif value == "" or value not in locales:
+        return default
```

Comment:
In order to avoid duplicating the list of languages here, maybe we can just check if the file exists rather than maintaining a separate list. Feel free to let me know what you think.

I also think for this function the `required` parameter isn't really necessary. In the other functions it was used since they could be reused for other things in the future that are required, but this probably won't be. It may be good to refactor the other functions at some point (in another PR).

```suggestion
    # if there is no yaml file for the language, use the default
    if not os.path.isfile(f"./locale/{value}.yml"):
        return default
```

Reply:
Checking for the file would be more maintainable than the list of locales 👍